src/App.tsx

```diff
@@ -21,6 +31,11 @@ export const App: React.FC = () => {
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            value={perPage}
+            onChange={(e) => {
+              setPerPage(+e.target.value);
+              setCurrentPage(1);
+            }}
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**